### ISSUE TITLE
[Jira] Refresh Token Fails with SSL (EPROTO) Error

### ISSUE THREAD

**Issue by macjabeth** [raycast/extensions#8696]
### Extension

https://www.raycast.com/raycast/jira

### Description

Yesterday, I was able to connect successfully without any problems after reinstalling and authenticating.

Today, when attempting to load "My Filters", I receive this failed authentication message.

**Error:**

![2023-10-05 at 10 00 49@2x](https://github.com/raycast/extensions/assets/8473052/5cee6297-9569-4475-b309-bb3fab579dd4)

### Steps To Reproduce

1. Install and authenticate Jira extension.
2. Wait until a token refresh is needed.
3. See failure.

### Current Behaviour

Token refresh is failing.

### Expected Behaviour

Token refresh should succeed.

**Comment by macjabeth** [comment]
This update comes after my prior update in #7335. I've tried reinstalling, revoking permissions in Jira, resetting the PEM file, and readding it. However, none of these attempts have been successful and the extension remains non-functional.

**Comment by macjabeth** [comment]
Worth noting that the Raycast Extension in Jira changed. Previously it was showing the correct logo, whereas now it is showing a grey icon. Not sure if it's related to the issue.

![2023-10-06 at 10 40 02@2x](https://github.com/raycast/extensions/assets/8473052/feafe79d-36f8-4b37-a8ec-6c3a8214801b)

**Comment by macjabeth** [comment]
After a few more attempts today, I managed to get it working again. I switched from using a custom pem to my keychain and had to restart Raycast and run the command twice for it to load correctly with the filters. It appears to be functional, although there are occasional issues that can be resolved by restarting or retrying the command a few times.

**Comment by rshkv** [comment]
For what it's worth, I'm also getting this with the Spotify extension when trying to authenticate:
```
ve: request to https://accounts.spotify.com/api/token failed, reason: write EPROTO 00B00F7201000000:error:0A000152:SSL routines:final_renegotiate:unsafe legacy renegotiation disabled:../deps/openssl/openssl/ssl/statem/extensions.c:922:


ClientRequest.<anonymous>:index.js:108:10

---
105: 		};
106: 
107: 		request_.on('error', error => {
108: 			reject(new FetchError(`request to ${request.url} failed, reason: ${error.message}`, 'system', error));
109: 			finalize();
110: 		});
111: 
---

ClientRequest.emit:node:events:525:35
ClientRequest.emit:node:domain:489:12
 ```

I've updated Raycast settings to trust my company's self-signed certificate.

**Comment by macjabeth** [comment]
I have additional findings to share. The connection is successful when I'm in the office and connected directly through the company's internal gateway. However, it fails when I try to connect from home while working remotely. This likely resolved the original problem when I closed issue #7335, along with the update to using the keychain in the advanced settings.

However, ideally I would like to connect to it remotely, not only within the office.

**Comment by rshkv** [comment]
In my case it's a corporate VPN that doesn't work with OpenSSL 3. There's an `UnsafeLegacyRenegotiation` option you can set in `openssl.cnf` to re-enable legacy negotiation and that's the fix behind our corporate VPN for other OpenSSL clients. But I can't figure out which `openssl.cnf` is used in Raycast.
